cpp/frontend_pyreadline.cc

```diff
@@ -235,4 +242,50 @@ Readline* MaybeGetReadline() {
 #endif
 }
 
+static void readline_cb(char* line) {
+  if (line == nullptr) {
+    gReadline->latest_line_ = nullptr;
+  } else {
+    gReadline->latest_line_ = StrFromC(line);
+    free(line);
+  }
+  gReadline->ready_ = true;
+  rl_callback_handler_remove();
+}
+
+Str* readline(Str* prompt) {
```

Comment:
Probably should point to these docs here:

https://tiswww.case.edu/php/chet/readline/readline.html#Alternate-Interface-Example